### DISCUSSION TITLE
There is a way to get the number of client connections from Node.js?

### DISCUSSION THREAD

**Discussion by deyveloper** [uNetworking/uWebSockets.js#1014] [Q&A] [answered]
I'm trying to build Prometheus metrics abstraction. I can't find a way to retrieve the client sockets connections count from uws application.

**Comment by uNetworkingAB** [comment]
App.filter lets you count them

**Comment by deyveloper** [comment]
![Screenshot 2024-01-29 at 13 44 22](https://github.com/uNetworking/uWebSockets.js/assets/62200663/e585019f-b699-455a-b237-b4ebf0c3be62)
Typescript error (property doesn't exists)

**Comment by deyveloper** [comment, marked as answer]
I upgraded to the latest version, and the filter worked, but there is a problem when the client ends the websocket connection (for example, closes the browser window) the callback is not called with a count value of -1, nothing is called, only the app close handler. But the filter for HTTP connections works fine, I'm receiving both counts 1, -1.

**Reply by uNetworkingAB** [reply]
Ah, that makes sense. It only tracks HTTP connections, if you upgrade to websocket you need to decrement it on .close. That's shitty but that's how it works right now (at least that behavior is stable)

**Reply by uNetworkingAB** [reply]
Simplest solution would be to emit a -1 on upgrade, then it would be logical for tracking HTTP and then you can track websockets on your own with open/close. But that is slightly API breaking so need to consider it.

**Reply by deyveloper** [reply]
I understand you, thank you for your promptness. By the way, we have been using your library for almost 4 years, we are very happy with it)
